x-pack/plugins/ml/server/lib/spaces_utils.ts

```diff
@@ -5,35 +5,28 @@
  * 2.0.
  */
 
-import { Legacy } from 'kibana';
 import { KibanaRequest } from '../../../../../src/core/server';
 import { SpacesPluginStart } from '../../../spaces/server';
 import { PLUGIN_ID } from '../../common/constants/app';
 
-export type RequestFacade = KibanaRequest | Legacy.Request;
-
 export function spacesUtilsProvider(
   getSpacesPlugin: (() => Promise<SpacesPluginStart>) | undefined,
-  request: RequestFacade
+  request: KibanaRequest
```

Comment:
There shouldn't be any way to access a legacy request from plugin code now, so I adapted the code to only handle `KibanaRequest`.

Reply:
Thanks. we can take care of others in https://github.com/elastic/kibana/issues/77997